### DISCUSSION TITLE
Is it not very accurate to use the size reward of the entire file as the reward for each caller-callee feature, if the file is large and has a large number of caller-callee?

### DISCUSSION THREAD

**Discussion by 18liumin** [google/ml-compiler-opt#356] [General]
def _overwrite_trajectory_reward(sequence_example: tf.train.SequenceExample,
                                 reward: float) -> tf.train.SequenceExample:
  """Overwrite the reward in the trace (sequence_example) with the given one.

  Args:
    sequence_example: A tf.SequenceExample proto describing compilation trace.
    reward: The reward to overwrite with.

  Returns:
    The tf.SequenceExample proto after post-processing.
  """
  sequence_length = len(
      next(iter(sequence_example.feature_lists.feature_list.values())).feature)

  reward_list = sequence_example.feature_lists.feature_list['reward']
  for _ in range(sequence_length):
    added_feature = reward_list.feature.add()
    added_feature.float_list.value.append(reward)

  return sequence_example

**Comment by mtrofin** [comment]
Notice the [discount factor](https://github.com/google/ml-compiler-opt/blob/main/compiler_opt/rl/agent_config.py#L214) is 1, which should mean only one of the reward values would be picked. It's probably superfluous we set the others to be the same, and hurts readability (clearly) - makes for easier experimentation with other discount values.
